plugins/providers/frontier/client.go

```diff
@@ -232,155 +238,170 @@ func (c *client) GetOrganizations() ([]*Organization, error) {
+	body["roleId"] = role
 
-	endPoint := path.Join(projectsEndpoint, "/", resource.ID, "/", role)
-	req, err := c.newRequest(http.MethodPost, endPoint, body, "")
+	req, err := c.newRequest(http.MethodPost, createPolicyEndpoint, body, "")
 	if err != nil {
 		return err
 	}
 
-	var users []*User
 	var response interface{}
 	if _, err := c.do(req, &response); err != nil {
+		c.logger.Error("Failed to grant access to the user", "Users", userId, req.URL)
 		return err
 	}
 
-	if v, ok := response.(map[string]interface{}); ok && v[usersConst] != nil {
-		err = mapstructure.Decode(v[usersConst], &users)
-		if err != nil {
-			return err
-		}
-	}
-
-	c.logger.Info("Project access to the user,", "total users", len(users), req.URL)
+	c.logger.Info("Project access to the user", "Users", userId, req.URL)
 	return nil
 }
 
-func (c *client) GrantOrganizationAccess(resource *Organization, userId string, role string) error {
-	body := make(map[string][]string)
-	body["userIds"] = append(body["userIds"], userId)
-
-	endPoint := path.Join(organizationEndpoint, "/", resource.ID, "/", role)
-	req, err := c.newRequest(http.MethodPost, endPoint, body, "")
+func (c *client) GrantOrganizationAccess(resource *Organization, userId string, roleId string) error {
+	body := make(map[string]string)
+	body["roleId"] = roleId
+	body["resource"] = fmt.Sprintf("%s:%s", "app/organization", resource.ID)
+	body["principal"] = fmt.Sprintf("%s:%s", "app/user", userId)
 
+	req, err := c.newRequest(http.MethodPost, createPolicyEndpoint, body, "")
 	if err != nil {
 		return err
 	}
 
-	var users []*User
 	var response interface{}
 	if _, err := c.do(req, &response); err != nil {
+		c.logger.Error("Failed to grant access to the user", "Users", userId, req.URL)
 		return err
 	}
 
-	if v, ok := response.(map[string]interface{}); ok && v[usersConst] != nil {
-		err = mapstructure.Decode(v[usersConst], &users)
-		if err != nil {
-			return err
-		}
-	}
-
-	c.logger.Info("Organization access to the user,", "total users", len(users), req.URL)
+	c.logger.Info("Organization access to the user,", "Users", userId, req.URL)
 	return nil
 }
 
-func (c *client) RevokeTeamAccess(resource *Team, userId string, role string) error {
-	endPoint := path.Join(groupsEndpoint, "/", resource.ID, "/", role, "/", userId)
-	req, err := c.newRequest(http.MethodDelete, endPoint, "", "")
+func (c *client) RevokeGroupAccess(resource *Group, userId string, role string) error {
+	endpoint := createPolicyEndpoint + "?groupId=" + resource.ID + "&userId=" + userId + "&roleId=" + role
+	req, err := c.newRequest(http.MethodGet, endpoint, "", "")
 	if err != nil {
 		return err
 	}
 
-	var success successAccess
+	var policies []*Policy
 	var response interface{}
 	if _, err := c.do(req, &response); err != nil {
 		return err
 	}
 
 	if v, ok := response.(map[string]interface{}); ok && v != nil {
-		err = mapstructure.Decode(v, &success)
+		err = mapstructure.Decode(v[policiesConst], &policies)
```

Comment:
what's the benefit of passing the `response` as `interface{}` first and then doing `Decode` again instead of directly decoding it in the actual type?